src/utils.js

```diff
@@ -38,6 +43,44 @@ function _normalizeUrl(src, baseUrl) {
 }
 module.exports._normalizeUrl = _normalizeUrl;
 
+async function _fetch(url) {
+  const res = await fetch(url)
+  if (res.status >= 200 && res.status < 300) {
+    return await res.text();
+  } else {
+    throw new Error('fetch got invalid status code: ' + res.status + ' : ' + url);
+  }
+}
+module.exports._fetch = _fetch;
+
+var _G = typeof global !== 'undefined' ? global : 
+         typeof window !== 'undefined' ? window :
+         typeof self !== 'undefined' ? self : this;
+var _context = vm.createContext(_G);
+
+async function _runMJS(code, {context, ...opts} = {}, fetch = _fetch) {
+  if (!vm.SourceTextModule) {
+    throw new Error('ECMAScript modules are not supported: ' + url);
+  }
+  if (!context) {
+    context = _context;
+  }
+  const script = new vm.SourceTextModule(code, {context, ...opts});
+  await script.link(async (path, {url, context}) => {
+    url = _normalizeUrl(path, url);
+    const code = await fetch(url);
+    return new vm.SourceTextModule(code, {url, context});
+  });
+  script.instantiate();
+  return await script.evaluate();
+}
+module.exports._runMJS = _runMJS;
```

Comment:
I'm not sure if `_runMJS` makes sense as a util, especially since it's dealing in global DOM things (capturing the global above). I would have put this in the DOM.js where its friends are.

Reply:
I can move it.

Reply:
Resolved in https://github.com/exokitxr/exokit/pull/1246.